File: golang/sequential/sequential.go
package sequential

import (
	"time"

	"github.com/MicaTravica/ntp/golang/savematrix"
	"github.com/MicaTravica/ntp/golang/util"
)

func multiply(mtx1 [][]float64, mtx2 [][]float64, n int) [][]float64 {
	mtx3 := make([][]float64, n)
	for i := range mtx3 {
		mtx3[i] = make([]float64, n)
		for j := 0; j < n; j++ {
			mtx3[i][j] = mtx1[i][j] * mtx2[i][j]
		}
	}
	return mtx3
}

// Sequential matrix multiplication
func Sequential(n int) {

	matrix1, matrix2 := util.CreateMatrix2(n)

	saveMatrix := savematrix.SaveMatrix{}
	saveMatrix.CrateSaveMatrix(matrix1, matrix2, 1)

	start := time.Now()

	for i := 0; i < n; i++ {
		matrix1[i] = append(matrix1[i][i:], matrix1[i][:i]...)
		a := make([]float64, n)
		for j := 0; j < n; j++ {
			a[j] = matrix2[(j+i)%n][i]
		}
		for j := 0; j < n; j++ {
			matrix2[j][i] = a[j]
		}
	}

	resultMatrix := multiply(matrix1, matrix2, n)
	saveMatrix.AddToCSequential(resultMatrix, 0)

	for s := 1; s < n; s++ {
		for i := 0; i < n; i++ {
			matrix1[i] = append(matrix1[i][1:], matrix1[i][:1]...)
		}
		matrix2 = append(matrix2[1:], matrix2[:1]...)
		util.AddAndMultiply(resultMatrix, matrix1, matrix2, n)
		saveMatrix.AddToCSequential(resultMatrix, s)
	}

	elapsed := time.Since(start)
	util.PrintResult(resultMatrix, elapsed)
	saveMatrix.WriteToFile()
}

File: python/sequential.py
import time
from util import create_2matrix, add_and_multiply, print_result
from save_matrix import SaveMatrix


def multiply(mtx1, mtx2, m_size):
    mtx3 = [[0] * m_size for _ in range(m_size)]
    for m_i in range(0, m_size):
        for m_j in range(0, m_size):
            mtx3[m_i][m_j] = mtx1[m_i][m_j] * mtx2[m_i][m_j]
    return mtx3


def sequential(size):
    matrix1, matrix2 = create_2matrix(size)
    save_matrix = SaveMatrix(matrix1, matrix2)

    start_time = time.time()
    # inicijalizacija matrica
    for i in range(1, size):
        matrix1[i] = matrix1[i][i:] + matrix1[i][:i]

        n = [matrix2[(j + i) % size][i] for j in range(0, size)]
        for j in range(0, size):
            matrix2[j][i] = n[j]

    result_matrix = multiply(matrix1, matrix2, size)
    save_matrix.add_to_c_sequential(result_matrix, True)

    for _ in range(1, size):
        for i in range(0, size):
            matrix1[i] = matrix1[i][1:] + matrix1[i][:1]
        matrix2 = matrix2[1:] + matrix2[:1]
        add_and_multiply(result_matrix, matrix1, matrix2, size)
        save_matrix.add_to_c_sequential(result_matrix)

    end_time = time.time()
    print_result(result_matrix, end_time - start_time)

    save_matrix.to_json_file()
